cpp/include/cuspatial/utility/vec_2d.cuh

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include <cuspatial/types.hpp>
+
+namespace cuspatial {
+
```

Comment:
I think the definition of the 2D vector types should be moved into this header as well. (types.hpp might go away...)

Also since this header can be used from host source files, it should use the .hpp extension.